cadence/server/db_redis.go

```diff
@@ -58,6 +64,53 @@ func rateLimit(next http.Handler) http.Handler {
 	})
 }
 
+func rateLimitArt(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		ip, err := checkIP(r)
+		if err != nil {
+			clog.Error("rateLimit", "Error encountered while checking IP address.", err)
+			w.WriteHeader(http.StatusInternalServerError) // 500 Internal Server Error
+			return
+		}
+		_, err = dbr.RateLimitArt.Get(ctx, ip).Result()
+		if err != nil {
+			if err == redis.Nil {
+				// redis.Nil means the IP is not in the database.
+				// We create a new entry for the IP with start value 1,
+				// representing the first request for art.
+				dbr.RateLimitArt.Set(ctx, ip, 1, time.Duration(200)*time.Second)
+				next.ServeHTTP(w, r)
+			} else {
+				clog.Error("rateLimit", "Error while attempting to check for IP in rate limiter.", err)
+				w.WriteHeader(http.StatusInternalServerError) // 500 Internal Server Error
+				return
+			}
+		} else {
+			// If there is no error, the IP is at least in the database.
+			// Check the value of the IP address.
+			count, err := dbr.RateLimitArt.Get(ctx, ip).Int()
+			if err != nil {
+				clog.Error("rateLimit", "Error while converting art served value to integer.", err)
+				w.WriteHeader(http.StatusInternalServerError) // 500 Internal Server Error
+				return
+			}
+			// If the IP has requested artwork at least 16 times, notify no-change.
+			// Basically, this response says "You've requested artwork a bit too much,
+			// so we're not going to send you new artwork for now. The artwork hasn't changed
+			// since you last asked, so you're safe to use whatever you last cached."
```

Comment:
If we're going to avoid adding the option to do a "Am I up to date" request, I suggest we leave a comment clarifying why we wait 16 times to send this.

IE - To allow clients which legitimately lose their caches (web) to still get art, while capping the maximum network they can burn (because we don't actually know the state of their cache, just asserting that "You've asked 15 times and got the same answer, you better have it cached!")

Just so if someone reads it in the future they know our justification for this - I can imagine someone reading this and asking why it's not one per song (as we originally had discussed)